### PR TITLE
Fortis: Adding Verify + Reference Purchase

### PR DESCRIPTION
**Pull request by Heavyblade**
## Summary:

Add neded changes to support reference authorize/purchase + supporting location_id as GSF for store

### Remote Test:
Finished in 181.033884 seconds.
25 tests, 58 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### Unit Tests:
Finished in 49.78482 seconds.
6143 tests, 80932 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed

### RuboCop:
806 files inspected, no offenses detected

### CODE REVIEW

**Review by naashton — APPROVED** [review]
LGTM!

File: lib/active_merchant/billing/gateways/fortis.rb [hunk 1]
```diff
@@ -175,11 +182,15 @@ def path(action, value = '')
...
           store: '/tokens/cc',
           unstore: '/tokens/{placeholder}'
-        }[action]&.gsub('{placeholder}', value.to_s)
+        }[action]&.gsub('{placeholder}', value.to_s.split('|').first || '')
       end
 
       def payment_type(payment)
-        payment.is_a?(String) ? 'token' : 'keyed'
+        if payment.is_a?(String)
+          payment.split('|').last == 'txn' ? 'prev-trxn' : 'token'
```
Comment (javierpedrozaing) [inline lib/active_merchant/billing/gateways/fortis.rb:190]: it took me a little bit to understand why the split is needed, but I can see that it is required for case like this. `"transaction/31e59fb2ec9d4bc5fb|txn/refund"` 👍🏼

File: test/remote/gateways/remote_fortis_test.rb [hunk 2]
```diff
@@ -155,6 +179,14 @@ def test_storing_credit_card
     assert_success store
   end
 
+  def test_storing_credit_card_with_location_as_option
```
Comment (javierpedrozaing) [inline test/remote/gateways/remote_fortis_test.rb:182]: is there another case where the `location_id` is required ? or maybe would be better to remove it from the fixtures ?

Comment (Heavyblade) [reply test/remote/gateways/remote_fortis_test.rb:182]: You can have location_id, in your fixtures/gateway optionally, but if you don't want to, then fo the store operation yopu need to send it as aGSF

**Pull request merged**